test/middlewares_test.exs

```diff
@@ -0,0 +1,179 @@
+defmodule MiddlewaresTest do
+  use ExUnit.Case
+
+  alias Exq.Middlewares.Server, as: Middlewares
+  alias Exq.Worker.Server, as: Worker
+
+  defmodule NoArgWorker do
+    def perform do
+    end
+  end
+
+  defmodule MissingMethodWorker do
+  end
+
+  defmodule MyMiddleware do
+    @behaviour Exq.Middlewares.Behaviour
+
+    import Exq.Middlewares.Pipeline
+
+    def before_work(pipeline) do
+      send :middlewarestest, :before_work
+      pipeline = assign(pipeline, :process_info, 1)
+      pipeline
+    end
+
+    def after_processed_work(pipeline) do
+      send :middlewarestest, :after_processed_work
+      pipeline
+    end
+
+    def after_failed_work(pipeline) do
+      send :middlewarestest, :after_failed_work
+      pipeline
+    end
+  end
+
+  defmodule HaltedMiddleware do
+    @behaviour Exq.Middlewares.Behaviour
+
+    import Exq.Middlewares.Pipeline
+
+    def before_work(pipeline) do
+      send :middlewarestest, :before_work_halted
+      halt(pipeline)
+    end
+
+    def after_processed_work(pipeline) do
+      send :middlewarestest, :after_processed_work_halted
+      halt(pipeline)
+    end
+
+    def after_failed_work(pipeline) do
+      halt(pipeline)
+    end
+  end
+
+  defmodule LoggerMiddleware do
```

Comment:
Awesome!  Can we make this a module outside the test directory? I was thinking this would be great to have as actual middleware to use!